### PR TITLE
Add support for expressions of type VariableRef

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -392,15 +399,22 @@ function parser_linear_expression(cmodel, nvar, F)
         end
         push!(constants, expr.constant)
       end
+      if isa(expr, VF)
```
Comment (tmigot) [inline src/utils.jl:402]: Can you add an example/unit test for this?

File: src/utils.jl [hunk 2]
```diff
@@ -409,6 +423,13 @@ function parser_linear_expression(cmodel, nvar, F)
...
```
Comment (tmigot) [inline src/utils.jl:426]: Same here

File: src/utils.jl [hunk 3]
```diff
@@ -25,6 +25,13 @@ const VI = MOI.VariableIndex
 const SQF = MOI.ScalarQuadraticFunction{Float64}
 const OBJ = Union{VI, SAF, SQF}
 
+# Expressions
```
Comment (tmigot) [inline src/utils.jl:28]: I don't know how useful that is, but why not.

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks for the changes. I think it lacks an example.

File: src/utils.jl (hunk 1 above)
Comment (amontoison) [reply src/utils.jl:402]: We already have example/unit test for this: https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/blob/main/test/nls_problems/hs30.jl#L6
I don't understand why the tests passed before.

File: src/utils.jl (hunk 2 above)
Comment (amontoison) [reply src/utils.jl:426]: We also need a NLPModel to verify the results ?

File: src/utils.jl (hunk 3 above)
Comment (amontoison) [reply src/utils.jl:28]: It helps me to remember the type of the objects used in MOI.

File: src/utils.jl (hunk 1 above)
Comment (tmigot) [reply src/utils.jl:402]: But according to `codecov` the tests are still not passing here

**Review by tmigot — APPROVED** [review]
LGTM, thanks!

**Comment by amontoison** [comment]
We can do a new release now.
@tmigot `v12.0` or `v11.2`?

**Comment by tmigot** [comment]
v12.0 because we drop JuMP 1.0 and 1.1

**Comment by tmigot** [comment]
Thanks!

**Pull request merged**